src/librustdoc/clean/mod.rs

```diff
@@ -80,12 +80,6 @@ impl<T: Clean<U>, U> Clean<U> for Rc<T> {
     }
 }
 
-impl<T: Clean<U>, U> Clean<Option<U>> for Option<T> {
-    fn clean(&self, cx: &mut DocContext<'_>) -> Option<U> {
-        self.as_ref().map(|v| v.clean(cx))
-    }
-}
```

Comment:
Why this change?

Reply:
I'd rather be more explicit that the code is dealing with an `Option`. I find the whole `Clean` trait quite confusing, and this way it's a bit simpler IMO. The impl was only used in a few places too.

If you want me to put it back, I can, but I think it's clearer to use `map` at the call site.

Reply:
I think as long as `Clean` is a trait it makes sense to keep the impls on wrapper types (the vec diff especially was super big and didn't seem particularly illuminating). If you want to make a PR at some point removing clean, I would be ok with removing these then :)

Reply:
> If you want to make a PR at some point removing clean, I would be ok with removing these then :)

The problem there is that removing Clean all at once would be a huge PR. This is a small step that would make a future removal more feasible.

> (the vec diff especially was super big and didn't seem particularly illuminating)

Also, see my comments about hidden performance issues caused by the Vec impl below :)